src/date-picker/__tests__/date-picker.test.tsx

```diff
@@ -193,23 +193,23 @@ describe('Date picker - direct date input', () => {
         value: '2018-03-01',
         onBlur: onBlurSpy,
       }));
-      act(() => wrapper.focus());
+      wrapper.focus();
     });
 
     afterEach(() => {
       jest.restoreAllMocks();
     });
 
     test('should fire blur handler when focusing another element', () => {
-      act(() => getByTestId(outsideId).focus());
+      getByTestId(outsideId).focus();
       expect(onBlurSpy).toHaveBeenCalledTimes(1);
     });
 
     test('should fire blur handler on open calendar when focusing another element', () => {
       wrapper.findOpenCalendarButton().click();
       expect(wrapper.findCalendarDropdown()).not.toBeNull();
 
-      act(() => getByTestId(outsideId).focus());
+      getByTestId(outsideId).focus();
```

Comment:
It would be very interesting if it passes. `getByTestId` returns you the native element instance, so `focus()` will not have any act wrappers

Reply:
It passed 👀

Reply:
Do we trust that it passes or would this be introducing something flaky?

Reply:
This test passing means, we do not have any `useEffect` depending on this focus event. If there will be any effects, they will not fire and you cannot write the proper assertion.

`act()` forces pending effects to fire, that's the whole trick